taichi/program/async_engine.cpp

```diff
@@ -264,50 +264,47 @@ TaskMeta AsyncEngine::create_task_meta(const TaskLaunchRecord &t) {
     if (auto global_load = stmt->cast<GlobalLoadStmt>()) {
       if (auto ptr = global_load->ptr->cast<GlobalPtrStmt>()) {
         for (auto &snode : ptr->snodes.data) {
-          meta.input_states.emplace_back(snode, AsyncState::Type::value);
+          meta.input_states.insert({snode, AsyncState::Type::value});
```

Comment:
nit: for consistency..

```suggestion
          meta.input_states.emplace(snode, AsyncState::Type::value);
```